sdk/storage/azure-storage-file-datalake/tests/test_file.py

```diff
@@ -1400,6 +1400,43 @@ def test_read_file_read(self, **kwargs):
         # Assert
         assert result == data
 
+    @DataLakePreparer()
+    @recorded_by_proxy
+    def test_create_and_read_file_encryption_context(self, **kwargs):
+        datalake_storage_account_name = kwargs.pop("datalake_storage_account_name")
+        datalake_storage_account_key = kwargs.pop("datalake_storage_account_key")
+
+        # Arrange
+        url = self.account_url(datalake_storage_account_name, 'dfs')
+        self.dsc = DataLakeServiceClient(url, credential=datalake_storage_account_key)
+        self.file_system_name = self.get_resource_name('filesystem')
+        file_name = 'testfile'
+        file_system = self.dsc.get_file_system_client(self.file_system_name)
+        try:
+            file_system.create_file_system()
+        except:
+            pass
+        file_client = file_system.get_file_client(file_name)
+
+        # Act
+        file_client.create_file(encryption_context='encryptionContext')
```

Comment:
Want to call out here that if you create a file with `encryption_context` set, then all the following asserts will pass. However, from further testing, I am unsure about expected behavior for the following scenarios:

Assume `data = b'Hello world'`

1. `file_client.upload_data(data)` is met with:
![image](https://user-images.githubusercontent.com/101599632/212217710-d22b431d-604b-42cb-bf42-7cd2d085e8a6.png)

2.  file_client.upload_data(data, overwrite=True)`, the following asserts will fail (seems like the overwrite gets rid of the encryption context information)

So in the first use case, how are customers expected to interact with a file that was created with an encryption context? In the second case, is this expected behavior (I guess knowing the answer to the first use case would imply an answer for the second)